### ISSUE TITLE
The devfile registry should allow referencing external devfiles or external git projects (that may contain devfile.yaml)

### ISSUE THREAD

**Issue by sunix** [eclipse-che/che#16377]
### Is your enhancement related to a problem? Please describe.
At the moment, the devfile registry can only use self referenced devfiles ( referenced in the same own repo). But for some team it may be hard to maintain these devfile in a single git repo. It may also be easier to maintain the devfile.yaml that is in the project repo it self.

### Describe the solution you'd like
in the devfile meta.yaml of a devfile registry, we should be able to indicate an external url of 

- a raw devfile
- a github project
- a git project

Basically all that the factory service is supporting with the url attribute.

**Comment by l0rd** [comment]
@sunix when you say "For some team it may be hard to maintain these devfile in a single git repo" what are you thinking about? How does teams are maintaining a devfile if not in a git repo? 

I am asking because even in the scenario that you are describing the team will need to:
- publish the devfile somewhere (and hopefully version it as well)
- maintain (create, update) the devfile meta.yaml in a devfile registry git repo
and that doesn't look simpler then maintaining the devfile.yaml in the git repo itself.

**Comment by sunix** [comment]
Take the example of a big project ... like Che:
you have
- the platform team
- the deploy team
- the plugin team
etc...
and they all share the same devfile registry. For all the git project they are owning, there are different permission policy: reviewers/commiters roles are different from one git repo to another.

But if you share the same one for devfile registry git how do you deal with all the different devfiles that should be own by a specific team ? It will be a nightmare to all the developers in the processes of reviewing/pushing/changing a devfile in the registry git repo.

Some teams would prefer to update their devfile that would be located in their own repo where they work. They would have the same permission and role regarding push/pull/review on the repo.
In the current situation, if a team would like to modify a devfile, they should go through the policy of the devfile registry git repo ... this is really annoying.

The meta.yaml file could just be created once and update not very often. No need to touch it if you just update the devfile. 

Also changing the devfile in your own git repo would not required to redeploy the devfile registry.

**Comment by l0rd** [comment]
The devfile registry is for sample stacks, not as a registry of teams projects (in those cases the devfile lives in the repository root folder).
And anyway, in the hypothetical case that different teams are responsible for different sample stacks, the `OWNERS` file you let you specify distinct ownership, even with a single GitHub repository.

**Comment by sunix** [comment]
What if teams are using Bitbucket and not Github? there is no such `OWNERS` things ...
 
If devfile registry is just for sample stacks, why do we make it such a big piece of our architecture. Why do we allow teams to provide their own devfile registry? Che should be the `Next-Generation IDE for developer teams` so the devfile registries should be for team projects as soon as they are used by developer teams.

Sample stacks are useful only when Che is being evaluated. But once a organisation is choosing to adopt Che, the devfile registry is the good place to list all the projects to be used daily. As you say, the devfile would live in the repository root folder. That is the purpose of this proposal to allow referencing these ones.

**Comment by l0rd** [comment]
Your ownership problem can be solved with [multiple devfile registries](https://github.com/eclipse/che/pull/16399): one registry per team. 

I am not convinced about the value of having the Dashboard "Getting Started" populated with real projects instead of samples.

**Comment by sunix** [comment]
Closing in favor of https://github.com/eclipse/che/issues/16875
